src/fides/api/ops/schemas/saas/saas_config.py

```diff
@@ -214,8 +214,28 @@ class SaaSRequestMap(BaseModel):
 class ConsentRequestMap(BaseModel):
     """A map of actions to Consent requests"""
 
-    opt_in: Optional[SaaSRequest] = None
-    opt_out: Optional[SaaSRequest] = None
+    opt_in: Union[SaaSRequest, List[SaaSRequest]] = []
+    opt_out: Union[SaaSRequest, List[SaaSRequest]] = []
+
+    @validator("opt_in")
+    def validate_opt_in(
+        cls,
+        opt_in: Union[SaaSRequest, List[SaaSRequest]],
+    ) -> List[SaaSRequest]:
+        """Convert all opt in request formats to a list of opt in requests"""
+        if isinstance(opt_in, SaaSRequest):
+            return [opt_in]
+        return opt_in
+
+    @validator("opt_out")
+    def validate_opt_out(
+        cls,
+        opt_out: Union[SaaSRequest, List[SaaSRequest]],
+    ) -> List[SaaSRequest]:
+        """Convert all opt out request formats to a list of opt out requests"""
+        if isinstance(opt_out, SaaSRequest):
+            return [opt_out]
+        return opt_out
```

Comment:
Consider combining both of these into one validator, I think `pydantic` allows you to specify multiple fields in the decorate definition like:

```
    @validator("opt_in", "opt_out")
    def validate_list_field(
        cls,
        field_value: Union[SaaSRequest, List[SaaSRequest]],
    ) -> List[SaaSRequest]:
        if isinstance(field_value, SaaSRequest):
            return [field_value]
        return field_value
```

Reply:
thank you Sean, I hadn't used this variant before, either single validators, or root validators, this seems like a good use case for specifying multiple fields.